evm_arithmetization/src/keccak_sponge/keccak_sponge_stark.rs

```diff
@@ -651,6 +655,39 @@ impl<F: RichField + Extendable<D>, const D: usize> Stark<F, D> for KeccakSpongeS
                     - next_values.already_absorbed_bytes),
         );
 
+        // If the first padding byte is at the end of the block, then the block has a
+        // single padding byte
+        let has_single_padding_byte = local_values.is_final_input_len[KECCAK_RATE_BYTES - 1];
```

Comment:
A proper appropriate name for `is_final_input_len` might be `is_first_padding_byte`. Let me know if you want me to change it